src/Nethermind/Nethermind.Blockchain/ReadOnlyChain.cs

```diff
@@ -38,7 +38,7 @@ public class ReadOnlyChain
 
         public ReadOnlyChain(ReadOnlyBlockTree readOnlyTree,
             IBlockValidator blockValidator,
-            IRewardCalculator rewardCalculator,
+            Func<ITransactionProcessor, IRewardCalculator> rewardCalculatorFactory,
```

Comment:
code smell here, a function that create a reward calculator from transaction processor?